src/full/Agda/TypeChecking/Conversion.hs

```diff
@@ -1714,14 +1708,12 @@ equalSort s1 s2 = do
            -- If @b@ is dependent, then @piSort a b@ computes to
            -- @Setω@. Hence, if @s@ is definitely not @Setω@, then @b@
            -- cannot be dependent.
-        if | definitelyNotInf s         -> do
-               -- We force @b@ to be non-dependent by unifying it with
-               -- a fresh meta that does not depend on @x : a@
-               b' <- newSortMeta
-               addContext (x,a) $ equalSort b (raise 1 b')
-               funSortEquals s (getSort a) b'
-           -- Otherwise: postpone
-           | otherwise                  -> synEq (PiSort a bAbs) s
+        if definitelyNotInf s then (do
+            -- We force @b@ to be non-dependent by unifying it with
+            -- a fresh meta that does not depend on @x : a@
+            b' <- newSortMeta
+            addContext (x,a) $ equalSort b (raise 1 b')
+            funSortEquals s (getSort a) b') else synEq (PiSort a bAbs) s
```

Comment:
I strongly prefer the previous version (possibly because I was the one who wrote it).